web_ext/sseq_gui/interface/index.js

```diff
@@ -278,11 +278,19 @@ messageHandler.Error = m => {
 document.getElementById('json-upload').addEventListener('change', () => {
     const file = document.getElementById('json-upload').files[0];
     const fileReader = new FileReader();
+
     fileReader.onload = e => {
-        // Remove whitespace to shorten URL
-        const json = JSON.stringify(JSON.parse(e.target.result));
-        window.location = `?module_json=${encodeURIComponent(json)}`;
+        try {
+            // Attempt to parse JSON
+            const json = JSON.stringify(JSON.parse(e.target.result));
+            window.location = `?module_json=${encodeURIComponent(json)}`;
+        } catch (error) {
+            // If an error occurs, display an error message
+            alert('Invalid JSON file. Please upload a valid JSON file.');
+            console.error('Error parsing JSON:', error);
```

Comment:
`error.message` contains useful info about the error:
```
Expected ',' or ']' after array element in JSON at position 21 (line 1 column 22)
```
It would be good to add this to the alert. Some browsers put the stack trace into `error.message` so it won't be completely consistent but that's okay.